### PR TITLE
DAOS-8539 test: Add dmg, daos, daos_agent, daos_server version test

### PR DESCRIPTION
**Pull request by shimizukko**
Verify version number and the output string for the four tools.

Added version subcommand support for dmg and daos util files.

Skip-unit-tests: true
Test-tag-vm: pr version_number
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/control/version.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        output = daos_cmd.version().stdout.decode("utf-8")
+
+        # Verify that "daos version" is in the output.
+        if "daos version" not in output:
+            errors.append(
+                "daos version is not in the output! {}".format(output))
+
+        result = re.findall(r"daos version ([\d.]+)", output)
+        daos_version = result[0]
+        self.log.info("daos version = %s", daos_version)
```
Comment (daltonbohning) [inline src/tests/ftest/control/version.py:65]: Similar to `dmg version`, `result[0]` would raise an exception if not found. For this and subsequent `version` commands.

**Review by daltonbohning — CHANGES_REQUESTED** [review]

**Comment by daltonbohning** [comment]
This is similar to `test_passwdlessssh_versioncheck` from #7728. I wonder if there is any overlap or possibility of sharing/combining? @saurabhtandan @shimizukko

**Comment by shimizukko** [comment]
It looks like @saurabhtandan 's PR only checks dmg and daos_server and not daos or daos_agent. Also, it doesn't look like it's comparing the obtained versions. It only checks if the versions are obtained.

File: src/tests/ftest/control/version.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        errors = []
+
+        # Get RPM version.
+        rpm_command = "rpm -qa|grep daos-server"
+        output = run_pcmd(hosts=self.hostlist_servers, command=rpm_command)
+        self.log.info("RPM output = %s", output)
+        stdout = output[0]["stdout"][0]
+        self.log.info("RPM stdout = %s", stdout)
+        result = re.findall(
+            r"daos-server-[tests-|tests_openmpi-]*([\d.]+)", stdout)
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/control/version.py]: Thanks for letting me know.

File: src/tests/ftest/control/version.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        dmg_cmd = self.get_dmg_command()
+        output = dmg_cmd.version().stdout.decode("utf-8")
+
+        # Verify that "dmg version" is in the output.
+        if "dmg version" not in output:
+            errors.append("dmg version is not in the output! {}".format(output))
+
+        result = re.findall(r"dmg version ([\d.]+)", output)
+        dmg_version = result[0]
+        self.log.info("dmg version = %s", dmg_version)
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/control/version.py]: Thanks. Fixed it.

File: src/tests/ftest/control/version.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/control/version.py]: Thanks. Fixed it.

File: src/tests/ftest/control/version.py [hunk 4]
```diff
@@ -48,21 +49,26 @@ def test_version(self):
             errors.append("dmg version is not in the output! {}".format(output))
 
         result = re.findall(r"dmg version ([\d.]+)", output)
-        dmg_version = result[0]
-        self.log.info("dmg version = %s", dmg_version)
+        if not result:
+            errors.append("dmg version is not in the output! {}".format(output))
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/control/version.py]: The first error message says it couldn't find the string "dmg version" in the command output. The second error message says it couldn't obtain the version number with the regex; e.g., it contains alphabet, comma, dash, etc.
I updated the error messages.

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/control/version.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/tests/ftest/control/version.py]: I see, thanks!

**Review by dinghwah — APPROVED** [review]
LGTM

**Pull request merged**